packages/hash/frontend/src/components/GlideGlid/use-grid-tooltip.tsx

```diff
@@ -0,0 +1,111 @@
+import { DataEditorRef } from "@glideapps/glide-data-grid";
+import { Popover } from "@hashintel/hash-design-system";
+import { PopoverPosition, Typography } from "@mui/material";
+import _ from "lodash";
+import { bindPopover, usePopupState } from "material-ui-popup-state/hooks";
+import { RefObject, useCallback, useState } from "react";
+import { useWindowEventListener } from "rooks";
+import { GridTooltipManager } from "./use-grid-tooltip/grid-tooltip-manager";
+import {
+  GridTooltip,
+  TooltipCellProps,
+  UseGridTooltipResponse,
+} from "./use-grid-tooltip/types";
+
+export const useGridTooltip = (
+  gridRef: RefObject<DataEditorRef>,
+): UseGridTooltipResponse => {
+  const popupState = usePopupState({
+    variant: "popover",
+    popupId: "grid-tooltip",
+  });
+
+  // prevents tooltip to stay at cursor position while user scrolls vertically
+  useWindowEventListener("scroll", () => {
+    popupState.close();
+  });
+
+  const [gridTooltip, setGridTooltip] = useState<GridTooltip | null>(null);
+  const [tooltipPos, setTooltipPos] = useState<PopoverPosition>();
+
+  const showTooltip = useCallback<TooltipCellProps["showTooltip"]>(
+    (newTooltip) => {
+      if (!_.isEqual(gridTooltip, newTooltip)) {
+        setGridTooltip(newTooltip);
+      }
+
+      const bounds = gridRef.current?.getBounds(newTooltip.col, newTooltip.row);
+
+      if (!bounds) {
+        return;
+      }
+
+      popupState.open();
+
+      const left = bounds.x + newTooltip.iconX;
+      const top = bounds.y;
+
+      setTooltipPos((prev) => {
+        if (prev?.left === left && prev?.top === top) {
+          return prev;
+        }
+
+        return { left, top };
+      });
+    },
+    [popupState, gridTooltip, gridRef],
+  );
+
+  const hideTooltip = useCallback<TooltipCellProps["hideTooltip"]>(
+    (_col, _row) => {
```

Comment:
Prefixing with underscore usually means unused but these are both used – why are they prefixed?

Reply:
Forgot to remove them after some refactoring, thanks for catching!